pkg/tsdb/influxdb/response_parser.go

```diff
@@ -24,45 +25,61 @@ func init() {
 func (rp *ResponseParser) Parse(response *Response, query *Query) plugins.DataQueryResult {
 	var queryRes plugins.DataQueryResult
 
+	if response.Error != "" {
+		queryRes.Error = fmt.Errorf(response.Error)
+		return queryRes
+	}
+
+	frames := data.Frames{}
 	for _, result := range response.Results {
-		queryRes.Series = append(queryRes.Series, rp.transformRows(result.Series, queryRes, query)...)
-		if result.Err != nil {
-			queryRes.Error = result.Err
+		frames = append(frames, transformRows(result.Series, query)...)
+		if result.Error != "" {
+			queryRes.Error = fmt.Errorf(result.Error)
 		}
 	}
+	queryRes.Dataframes = plugins.NewDecodedDataFrames(frames)
 
 	return queryRes
 }
 
-func (rp *ResponseParser) transformRows(rows []Row, queryResult plugins.DataQueryResult, query *Query) plugins.DataTimeSeriesSlice {
-	var result plugins.DataTimeSeriesSlice
+func transformRows(rows []Row, query *Query) data.Frames {
+	frames := data.Frames{}
 	for _, row := range rows {
 		for columnIndex, column := range row.Columns {
 			if column == "time" {
 				continue
 			}
 
-			var points plugins.DataTimeSeriesPoints
+			var timeArray []time.Time
+			var valueArray []*float64
+
 			for _, valuePair := range row.Values {
-				point, err := rp.parseTimepoint(valuePair, columnIndex)
-				if err == nil {
-					points = append(points, point)
+				timestamp, timestampErr := parseTimestamp(valuePair[0])
+				// we only add this row if the timestamp is valid
+				if timestampErr == nil {
+					value, valueErr := parseValue(valuePair[columnIndex])
+					// if there is a value-error, we use nil as the value
+					if valueErr != nil {
+						value = nil
+					}
```

Comment:
I think when the `valueErr` is not nil, `value` will always be nil right? 

From what `parseValue(...)` returns - `return nil, fmt.Errorf("value has invalid type: %#v", value)` on line _173_ 
 and `return nil, err` on line _181_?

Reply:
@dsotirakis  hmm.. you are correct, but i would prefer to not rely on that...in my understanding, `parseValue`'s promise is this: "i get you the value or an error". so it feels to me i should not touch the returned-value if there is a returned-error. what do you think?

alternatively, i could just simplify things and make `parseValue` never return an error, only a `*float64`, and do this decision inside it.

Reply:
Maybe you could do the latter I guess since re-assigning a value to `nil` when is already `nil` for better visibility doesn't seem right to me! Maybe everything must happen inside `parseValue`